### ISSUE TITLE
[1.13] GuiLabel source doesn't match compiled class

### ISSUE THREAD

**Issue by skyecodes** [MinecraftForge/MinecraftForge#5704]
After I update the forge version in my dev environment and import the gradle project, I can see the decompiled GuiLabel.java file : https://pastebin.com/UwyGP6Mf
After refreshing the gradle project, I can see the source GuiLabel.java file : https://pastebin.com/qj0qHTR3

As you can see, there is no constructor in the decompiled class, whereas there's one in the source file. The GuiLabel class must have a constructor because there are final fields that aren't initialized.

It causes an issue in my mod because I use a subclass of GuiLabel. Whenever I try to create a new instance of this subclass and call the superclass constructor, I have an error.
My subclass (it's an inner class) : https://pastebin.com/bRp6v1iW
The error : https://pastebin.com/X5tu7Yav

MinecraftForge version : 25.0.146
Mappings : 20190417-1.13.2

PS : This GuiLabel class has so many private fields that we can't access / modify. Maybe why not add them to the Forge AT ?

**Comment by LexManos** [comment]
There are times where the initial IDE setup will have JUST the binary version of the class.
And the first one you're seeing is your IDE {Most likely IntelliJ} decompiling the class.
It uses an older version of the same decompiler we do. Which has.. issues...
We fix this one in particular with a patch: https://github.com/MinecraftForge/MCPConfig/blob/master/versions/1.13.2/patches/joined/net/minecraft/client/gui/GuiLabel.java.patch
This one specifically, is a issue with how Mojang ships the game. They actually strip out this constructor, because nowhere in Vanilla's codebase does it call this function.
Basically, just import the project twice to be sure you have the -recomp jar as a dependency. As that's the full decompiled/patched/recompiled version.

**Comment by LexManos** [comment]
Actually re-opening, as this class isn't patched by Forge. And thus the MCP'ed in constructor will not exist at runtime.
We need an excuse to patch this class so that it will be included in the binpatches and get the constructor.

**Comment by skyecodes** [comment]
I'm creating a custom GUI library to automatically handle widget positions using vertical and horizontal boxes. I rely on vanilla and Forge classes for rendering (GuiButtonExt, GuiCheckbox, GuiTextField and GuiLabel). It prevents me from copy-pasting rendering code that already exists.

But if you don't want to apply your GuiLabel patch (and I understand that, this class isn't used anywhere), then I'll just copy-paste the code from this class. It will give me more control on GuiLabel properties that were private and that I could only access using reflection.
